### ISSUE TITLE
Min fee estimation for a wallet is too high

### ISSUE THREAD

**Issue by noescape00** [stratisproject/StratisBitcoinFullNode#1283]
Wallet min fee estimation is ~60 times bigger than actual. 
With empty blocks the min fee to get into the block is: `tx serialized size / 100000000` (`PowBlockAssembler.cs:499`)

So for a tx that is 229 bytes fee is 0.00000229 and the wallet estimation algho says its 0.00013800

**Comment by majikandy** [comment]
![60x](https://user-images.githubusercontent.com/444711/38859865-d54b49d4-4226-11e8-9e4c-8e5e69eacd90.PNG)

Here we can see where the fallback fee of 60x the 0.00001 / kb    ->  0.0006 /kb

**Comment by majikandy** [comment]
In the call to wallet fee policy we can see that the smart fee isn't estimated as that code is commented out. And instead it uses a fallback value

```
WalletFeePolicy.cs
...
        public FeeRate GetFeeRate(int confirmTarget)
        {
            //this.blockPolicyEstimator.EstimateSmartFee(confirmTarget, this.mempool, out estimateFoundTarget).GetFee(txBytes);
            return this.fallbackFee;
        }
...
```
In discussions with @bokobza, this is a bigger piece of work for the fee estimator, so this fallback fee was chosen as something quite high to help make sure transactions using the wallet get included in a block soon.

Further discussion with @noescape00 suggested that since the blocks aren't currently full (which would be a more difficult calculation)  we could implement a simple algorithm of checking whether the last few blocks weren't full.

if prev n blocks were not full- use constant fee value of `txSizeKb * minFeePerKb` where minFeePerKb is the max value between current C# implementation and QT

**Comment by dangershony** [comment]
For now the default is fine for stratis (I don't think we need to check for full blocks) this is lower priority.
However the end goal is to port over the new fee estimator from bitcoin core (I ported over the old one and its not complete, that's why its commented out).

**Comment by noescape00** [comment]
The size estimation is for a quick fix which would be using min fee when the blocks are not full.

**Comment by mikedennis** [comment]
I assume this is only for fullnode wallet as light wallet does not have reference to blockstore. So only going to add this to walletpolicy for now which only applies to fullnode wallet.

**Comment by noescape00** [comment]
Yes. 

About light wallet we are not sure right now so let's not touch it yet.

**Comment by mikedennis** [comment]
So the fee estimation is done via the transaction builder (NBitcoin). It requires the minimum to be at least ```minRelayTxFee``` which is currently 0.0001 Strat/kb.  I'm not sure where the 60x comes from mentioned above in PowBlockAssembler.cs, as this file no longer seems to exist in the codebase, 

StratisX uses a fixed fee of 0.0001 (10,000 Satoshi) so the fallback fee that full node wallet is currently using should only be 6x what StratisX is doing not 60x. I don't think StratisX is changing rates based upon full blocks.

So I recommend either we:
1. Use the minRelayTxFee (0.0001) instead of fallbackfee. This would give us same behaviour as StratisX
2. If it is desirable to treat empty blocks differently then use minRelayTxFee (0.0001) for empty blocks and fallBackTxFee (0.0006) for non empty blocks.

I've already written the empty block logic so no problem doing it but seems to me a bit ugly having this conditional logic and I wonder whether we would be better off just using the same fixed fee as StratisX.

**Comment by bokobza** [comment]
I prefer having the same as StratisX.
I wonder where this 60K sats/Kb is coming from.
